pylint/checkers/classes/class_checker.py

```diff
@@ -49,19 +49,33 @@
 # Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 # For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
 
-"""classes checker for Python code
-"""
-import collections
-from itertools import chain, zip_longest
-from typing import List, Pattern
+"""classes checker for Python code """
+
+
+from itertools import chain
+from typing import List
 
 import astroid
 from astroid import nodes
 
 from pylint.checkers import BaseChecker, utils
+from pylint.checkers.classes.class_checker_helper import (
+    _ancestors_to_call,
+    _called_in_methods,
+    _definition_equivalent_to_call,
+    _has_bare_super_call,
+    _has_data_descriptor,
+    _has_different_parameters_default_value,
+    _has_same_layout_slots,
+    _is_attribute_property,
+    _is_invalid_base_class,
+    _signature_from_arguments,
+    _signature_from_call,
+)
+from pylint.checkers.classes.different_parameters import _different_parameters
+from pylint.checkers.classes.scope_access_map import ScopeAccessMap
```

Comment:
All these are only use in this file. I think it would be better to keep them together.
Otherwise we just end up with a lot of util files everywhere.

Reply:
Yeah I wondered about that too. There's a practical aspect I considered: the parsing time of the file when you work on it. The main pain point is the ClassChecker itself being so big though and creating util file does not solve the main design problem. I'll revert.

Reply:
At least in VS Code, I don't find the parsing time to be an issue.
Of course it takes a few seconds until the pylint warnings show up but that isn't a problem for me.

Reply:
For 2000+ line file Pycharm analysis takes more than 5s on my setup (which is not high end but not that bad either). I guess low budget work station would have even more lag with our biggest files.

Reply:
(Pycharm is a JetBrain IDE, not using pylint as linter)

Reply:
You should really try VS Code someday 😄 I use a few years old, 2 core MacBook Air.

Reply:
I'm also in a VM that do not have enough disk space 😄 Help to commiserate with developer without a good equipment :)